test/integration/cloudbuild_repo_connection_github/cloudbuild_repo_connection_github_test.go

```diff
@@ -0,0 +1,151 @@
+}
+
+func (gh *GitHubClient) GetRepository(ctx context.Context) *github.Repository {
+	repo, resp, err := gh.client.Repositories.Get(ctx, gh.owner, gh.repoName)
+	if resp.StatusCode != 404 && err != nil {
+		gh.t.Fatal(err.Error())
+	}
+	gh.repository = repo
+	return repo
+}
+
+func (gh *GitHubClient) CreateRepository(ctx context.Context, org, repoName string) *github.Repository {
+	newRepo := &github.Repository{
+		Name:       github.String(repoName),
+		AutoInit:   github.Bool(true),
+		Private:    github.Bool(true),
+		Visibility: github.String("private"),
+	}
+	repo, _, err := gh.client.Repositories.Create(ctx, org, newRepo)
+	if err != nil {
+		gh.t.Fatal(err.Error())
+	}
+	gh.repository = repo
+	return repo
+}
+
+func (gh *GitHubClient) DeleteRepository(ctx context.Context) {
+	resp, err := gh.client.Repositories.Delete(ctx, gh.owner, *gh.repository.Name)
+	if resp.StatusCode != 404 && err != nil {
+		gh.t.Fatal(err.Error())
+	}
+}
+
+func TestCloudBuildRepoConnectionGithub(t *testing.T) {
+	ctx := context.Background()
+
+	repoName := fmt.Sprintf("conn-gh-%s", utils.GetRandomStringFromSetup(t))
+
+	githubPAT := cftutils.ValFromEnv(t, "IM_GITHUB_PAT")
+
+	owner := "im-goose"
+	client := NewGitHubClient(t, githubPAT, owner, repoName)
+
+	repo := client.GetRepository(ctx)
+	if repo == nil {
+		client.CreateRepository(ctx, client.owner, client.repoName)
+	}
+
+	repoURL := client.repository.GetCloneURL()
+
+	resourcesLocation := "us-central1"
+	vars := map[string]interface{}{
+		"github_pat":      githubPAT,
+		"github_app_id":   "47590865",
+		"repository_name": repoName,
+		"repository_url":  repoURL,
+	}
+
+	bpt := tft.NewTFBlueprintTest(t, tft.WithVars(vars))
+
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		bpt.DefaultVerify(assert)
+
+		t.Cleanup(func() {
+			// Delete the repository if we hit a failed state
+			if t.Failed() {
+				client.DeleteRepository(ctx)
+			}
+		})
+
+		// validate if repository was created using the connection
+		projectId := bpt.GetTFSetupStringOutput("project_id")
+		connectionId := bpt.GetStringOutput("cloud_build_repositories_2nd_gen_connection")
+
+		connectionIdRegexPattern := `^projects/[^/]+/locations/[^/]+/connections/[^/]+$`
+		re := regexp.MustCompile(connectionIdRegexPattern)
+		assert.True(re.MatchString(connectionId), "Connection ID should be in format projects/{{project}}/locations/{{location}}/connections/{{name}}")
+
+		connectionSlice := strings.Split(connectionId, "/")
+		// Extract the project, location and connection name from the connection_id
+		connectionProjectId := connectionSlice[1]
+		connectionLocation := connectionSlice[3]
+		connectionName := connectionSlice[len(connectionSlice)-1]
+
+		// Assert that the resource was created in the specified project and region
+		assert.Equal(projectId, connectionProjectId, "Connection project id should be the same as input project id.")
+		assert.Equal(resourcesLocation, connectionLocation, "Connection location should be 'us-central'.")
```

Comment:
`us-central1`? Perhaps better to use `resourcesLocation` in `fmt.Sprintf`.

```suggestion
		assert.Equal(resourcesLocation, connectionLocation, "Connection location should be 'us-central`'.")
```